lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -550,8 +554,54 @@ def _graded_scorable_blocks_to_header(cls, course):
 
 
 class ProblemResponses(object):
+
+    @classmethod
+    def _build_problem_list(cls, course_blocks, root, path=None):
+        """
+        Generate a tuple of display names, block location paths and block keys
+        for all problem blocks under the ``root`` block.
+        """
+        if path is None:
+            path = [course_blocks.get_xblock_field(root, 'display_name')]
+        for block in course_blocks.get_children(root):
+            display_name = course_blocks.get_xblock_field(block, 'display_name')
+            if block.block_type == 'problem':
+                yield display_name, path, block
+            else:
+                for result in cls._build_problem_list(course_blocks, block,
+                                                      path + [display_name]):
```

Comment:
Nit: Don't squish params on the right hand side - http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines